docker/Rocky.dockerfile

```diff
@@ -0,0 +1,89 @@
+ARG rocky_version=9
+FROM rockylinux:${rocky_version} as common-base
+
+ENV TZ=America/Chicago
+RUN ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone
+RUN dnf update -y && \
+    dnf install -y 'dnf-command(config-manager)' && \
+    dnf install -y python3.11 epel-release
+
+FROM common-base as rocky-8-config
+RUN alternatives --set python /usr/bin/python3.11 && \
+    alternatives --set python3 /usr/bin/python3.11 && \
+    dnf config-manager --set-enabled powertools
+
+FROM common-base as rocky-9-config
+RUN alternatives --install /usr/bin/python python /bin/python3.11 10 && \
+    alternatives --install /usr/bin/python3 python3 /bin/python3.11 10 && \
+    dnf config-manager --set-enabled crb
+
+FROM rocky-${rocky_version}-config as dnf-deps
+RUN dnf install -y \
+        wget \
+        which \
+        git \
+        gcc \
+        gcc-c++ \
+        make \
+        cmake \
+        hdf5-devel \
+        libxml2-devel \
+        boost-devel \
+        blas-devel \
+        lapack-devel \
+        sqlite-devel \
+        gettext \
+        xz \
+        python3.11-devel \
+        python3.11-setuptools \
+        python3.11-pip \
+        python3.11-pytest \
+    && dnf clean all
+
+RUN mkdir -p $(python -m site --user-site) && python -m pip install pandas tables cython jinja2
+
+FROM dnf-deps as libxmlpp
+RUN dnf install -y m4 doxygen perl-open perl-XML-Parser diffutils pcre-cpp pcre-devel  && \
+    python -m pip install meson ninja packaging && \
+    wget https://github.com/libxmlplusplus/libxmlplusplus/releases/download/4.0.3/libxml++-4.0.3.tar.xz && \
```

Comment:
The RHEL world seems to be stubbornly opposed to updating this, huh?

Reply:
Yeah most of the packages on RHEL are very old.  This [website](https://packages.fedoraproject.org/pkgs/libxml++/libxml++/) is a good explorer if you're curious about any other package versions.